h/static/scripts/directive/test/annotation-test.js

```diff
@@ -470,6 +470,177 @@ describe('annotation.js', function() {
+          then: function() {}
+        });
+
+        createDirective();
+
+        assert.notCalled(annotation.$create);
+      });
+
+      it('does not save old highlights on initialization', function() {
+        // Old highlights (ones that the client has received from the server,
+        // rather than created locally itself) have an id and do not have any
+        // $highlight.
+        annotation.id = 'annotation_id';
+        annotation.$highlight = undefined;
+
+        // If it's a highlight, then the annotation will have a target but
+        // no references, text or tags.
+        annotation.target = ['foo', 'bar'];
+        annotation.references = [];
+        annotation.text = '';
+        annotation.tags = [];
+
+        annotation.$create = sandbox.stub().returns({
+          then: function() {}
+        });
+
+        createDirective();
+
+        assert.notCalled(annotation.$create);
+      });
+
+      it('does not save old annotations on initialization', function() {
+        // Old annotations (ones that the client has received from the server,
+        // rather than created locally itself) have an id and do not have any
+        // $highlight.
+        annotation.id = 'annotation_id';
+        annotation.$highlight = undefined;
+
+        // If it's an annotation (rather than a highlight, reply or page note)
+        // then the annotation will have a target, no references and some
+        // text or tags.
+        annotation.target = ['foo', 'bar'];
+        annotation.references = [];
+        annotation.text = 'This is my annotation';
+        annotation.tags = ['tag_1', 'tag_2'];
+
+        annotation.$create = sandbox.stub().returns({
+          then: function() {}
+        });
+
+        createDirective();
+
+        assert.notCalled(annotation.$create);
+      });
+
+      it('edits new annotations on initialization', function() {
+        // When the user creates a new annotation and we create a new
+        // AnnotationController instance, we automatically open the
+        // annotation's editor.
+
+        // A new annotation will have no id or $highlight.
+        annotation.id = annotation.$highlight = undefined;
+
+        // A new annotation won't have any text or tags yet.
+        annotation.text = '';
+        annotation.tags = [];
+
+        // A new annotation won't have any saved drafts yet.
+        fakeDrafts.get.returns(null);
+
+        var controller = createDirective().controller;
+
+        assert.isTrue(controller.editing());
+      });
+
+      it('edits annotations with drafts on initialization', function() {
+        // This is not a new annotation.
+        annotation.id = 'annotation_id';
+        // This is not a highlight.
+        annotation.$highlight = undefined;
+        annotation.text = 'Annotation text';
+        annotation.tags = ['tag_1', 'tag_2'];
+        // The drafts service has some draft changes for this annotation.
+        fakeDrafts.get.returns('foo');
```

Comment:
This should return an object that conforms to the interface for `drafts.get` (`{text?: string, tags?: string[], ...}`). It happens to work with the current implementation of AnnotationController, but could incorrectly break in future.